carla_ros_bridge/src/carla_ros_bridge/bridge.py

```diff
@@ -297,55 +307,66 @@ def publish_bboxes_and_lidar_overlay(self):
             self.logwarn("Lidar overlay.. couldnt setup lidar_to_rgb.")
 
         vehicles = self.carla_world.get_actors().filter('vehicle.*')
-        for id_, rgb_cam in self.rgb_cams.items():
-            bounding_boxes = ClientSideBoundingBoxes.get_bounding_boxes(vehicles, rgb_cam.carla_actor)
-            img = rgb_cam.get_image()
-            frame = rgb_cam.get_frame()
-            if img is None: continue
-
-            # draw bounding boxes
-            for bbox in bounding_boxes:
-                points = [(int(bbox[i, 0]), int(bbox[i, 1])) for i in range(8)]
-                # base
-                cv2.line(img, points[0], points[1], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[0], points[1], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[1], points[2], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[2], points[3], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[3], points[0], CarlaRosBridge.BB_COLOR, thickness=2)
-                # top
-                cv2.line(img, points[4], points[5], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[5], points[6], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[6], points[7], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[7], points[4], CarlaRosBridge.BB_COLOR, thickness=2)
-                # base-top
-                cv2.line(img, points[0], points[4], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[1], points[5], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[2], points[6], CarlaRosBridge.BB_COLOR, thickness=2)
-                cv2.line(img, points[3], points[7], CarlaRosBridge.BB_COLOR, thickness=2)
-
-            # lidar points
-            lidar_data = self.lidar.get_lidar_data()
-            if self.lidar_to_camera is not None:
-                if lidar_data is not None:
-                    if lidar_data.frame == frame:
-                        # draw
-                        image_w = float(rgb_cam.carla_actor.attributes.get("image_size_x"))
-                        image_h = float(rgb_cam.carla_actor.attributes.get("image_size_y"))
-                        img = self.lidar_to_camera.lidar_overlay(lidar_data, img, rgb_cam, image_w, image_h)
+        for parent, rgb_cams in self.rgb_cams.items():
+            # check if lidar exists for the same parent
+            curr_lidar = None
+            try:
+                curr_lidar = self.lidar[parent]
```

Comment:
Would be cleaner if you add a check instead:

```python
if parent in self.lidar:
  curr_lidar = self.lidar[parent]
else:
  self.node.logwarn(f"lidar is None for {parent}")
```

Reply:
yep right. thanks!